src/scroll-timeline-css.js

```diff
@@ -0,0 +1,192 @@
+import { StyleParser, removeKeywordsFromAnimationShorthand } from "./scroll-timeline-css-parser";
+
+const parser = new StyleParser();
+
+function initMutationObserver() {
+  const sheetObserver = new MutationObserver((entries) => {
+    for (const entry of entries) {
+      for (const addedNode of entry.addedNodes) {
+        if (addedNode instanceof HTMLStyleElement) {
+          handleStyleTag(addedNode);
+        }
+        if (addedNode instanceof HTMLLinkElement) {
+          handleLinkedStylesheet(addedNode);
+        }
+      }
+    }
+
+    // TODO: Not a good idea to proxy all elements.
+    // Not adding proxy affects tests that there is no animation,
+    // but there is change for 'animation', 'animation-timeline', etc via style
+    // like this: e.style['animation-timeline'] = "none, auto"
+    // document.querySelectorAll("body *").forEach(element => {
+    //   addProxyForElement(element);
+    // });
+  });
+
+  sheetObserver.observe(document.documentElement, {
+    childList: true,
+    subtree: true,
+  });
+
+  function handleStyleTag(el) { // el: HtmlStyleElement
+    // Don’t touch empty style tags.
+    if (el.innerHTML.trim().length === 0) {
+      return;
+    }
+    const newSrc = parser.transpileStyleSheet(el.innerHTML);
+    el.innerHTML = newSrc;
+  }
+
+  function handleLinkedStylesheet(el) {
+    // TODO
+  }
+
+  document.querySelectorAll("style").forEach((tag) => handleStyleTag(tag));
+  document
+    .querySelectorAll("link")
+    .forEach((tag) => handleLinkedStylesheet(tag));
+}
+
+function getSourceElement(source) {
```

Comment:
Worth adding a comment for which version of the spec this is based on as this is likely to change.  Add a TODO to update to new version of the spec.